x-pack/legacy/plugins/spaces/server/lib/utils/namespace.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { DEFAULT_SPACE_ID } from '../../../common/constants';
+
+export function spaceIdToNamespace(spaceId: string): string | undefined {
+  if (!spaceId) {
+    throw new TypeError('spaceId is required');
+  }
+
+  if (spaceId === DEFAULT_SPACE_ID) {
+    return undefined;
+  }
+
+  return spaceId;
+}
+
+export function namespaceToSpaceId(namespace: string | undefined): string {
+  if (!namespace) {
```

Comment:
Should we throw an error on namespace being `''` or else we risk returning undefined here...

Reply:
Are we only concerned about the empty string scenario because that's valid given the typedef, or should we be defending against other falsy values too?

Reply:
> Are we only concerned about the empty string scenario because that's valid given the typedef

Pretty much... Ideally, I think that anything that is exposed outside of the context of a plugin should be more defensive than we've traditionally been and perhaps use https://github.com/elastic/kibana/tree/master/packages/kbn-config-schema to validate all inputs. To keep this from getting too far out of scope, doing validation based on the valid type definitions seemed like a good place to stop for this PR.